lib/timers/promises.js

```diff
@@ -111,7 +111,86 @@ function setImmediate(value, options = {}) {
       () => signal.removeEventListener('abort', oncancel)) : ret;
 }
 
+async function* setInterval(after, value, options = {}) {
+  if (options == null || typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options',
+      'Object',
+      options);
+  }
+  const { signal, ref = true, throwOnAbort = true } = options;
+  validateAbortSignal(signal, 'options.signal');
+  if (typeof ref !== 'boolean') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options.ref',
+      'boolean',
+      ref);
+  }
+
+  if (typeof throwOnAbort !== 'boolean') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options.throwOnAbort',
+      'boolean',
+      ref);
+  }
+
+  if (signal?.aborted) {
+    if (throwOnAbort) throw new AbortError();
+    return;
+  }
+
+  let onCancel;
+  let notYielded = 0;
+  let passCallback;
+  let abortCallback;
+  const interval = new Timeout(() => {
+    notYielded++;
+    if (passCallback) {
+      passCallback();
+      passCallback = undefined;
+      abortCallback = undefined;
+    }
+  }, after, undefined, true, true);
+  if (!ref) interval.unref();
+  insert(interval, interval._idleTimeout);
+  if (signal) {
+    onCancel = () => {
+      // eslint-disable-next-line no-undef
+      clearInterval(interval);
+      if (abortCallback) {
+        abortCallback(new AbortError());
+        passCallback = undefined;
+        abortCallback = undefined;
+      }
+    };
+    signal.addEventListener('abort', onCancel, { once: true });
+  }
+
+  while (!signal?.aborted) {
+    if (notYielded === 0) {
```

Comment:
Can we get rid of the `if` statement by moving the `for` loop above the `await new Promise`?

Reply:
I think that we would get an issue like the following:
  suppose that the promise is yielded, and then nobody calls next for a while, getting "backpressure". The next time somebody calls next, we'll go back to the while condition and get out of the while loop, instead of yielding all of the missing values